### PR TITLE
Add `npm run x-compile` and `npm run x-diff` for visual test 

### LINKED ISSUE #437: Tools for Visual Test
## Gallery 
- [x] script for compiling example specs
- [x] corpus of example specifications
- [x] group them into different test set

## Visual Test
- [x] script for compiling all example specifications 
- [x] script for compare each compiled vega spec to its older version
- [x] script for showing all the specs that has been changed

### PR DESCRIPTION
**Pull request by kanitw**
Fixes #437 

Please merge #848 and #849 first!  
(See [Branch Specific Diff](https://github.com/uwdata/vega-lite/compare/kw/examples...kw/visual-test))

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -103,7 +111,15 @@ If you only want subset of these actions, you can use:
 
 - `npm run watch:build` to start a watcher task that **re-compiles Vega-Lite** when `.ts` files related to VL change.
 
+#### Output diff
+
+We also have commands for observing changes in output Vega spec and output images.
+
+To create a baseline Vega output specs from the Vega-Lite specs in `examples/`, check out the baseline branch (e.g., `git checkout master`) and run `npm x-compile`.
```
Comment (domoritz) [inline README.md]: remove a

File: scripts/examples-diff.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  name=${file##*/} # filename only (no dir path)
+  base=${name%.json} # exclude extension
+  if [ "$name" != "vlexamples.json" ]; then  # don't compile the list json
+    diff=$(json-diff --color examples/_original/$name examples/_output/$name)
+    if [ "$diff" != " undefined" ]; then
+      echo "Diff for $name"
+      echo -e "$diff"
+      changed+=("$base")
+      # compile each vega output to svg in examples/_diff
+      # and use "xmllint --format -" to format
```
Comment (domoritz) [inline scripts/examples-diff.sh]: I hope I don't have to look at too many xml outputs ;-)

Comment (kanitw) [inline scripts/examples-diff.sh]: Umm, most of the time, you should just compare rendered svg.  (Mac’s quicklook supports SVG I think)

**Comment by kanitw** [comment]
> Please undo since datalib fix has been released.
I don’t think you have to comment this 3 times.   It obviously depends on other PR.

File: scripts/examples-diff.sh (hunk 2 above)
Comment (domoritz) [inline scripts/examples-diff.sh]: Hopefully. Cool stuff btw.

**Comment by domoritz** [comment]
> I don’t think you have to comment this 3 times. It obviously depends on other PR.

I commented on the commit but since that commit is used in many PRs, it appears multiple times.

**Comment by domoritz** [comment]
LGTM, merge when you resolved conflicts with master.

**Pull request merged**